motr/m0crate/btree_without_tx_seg.c

```diff
@@ -490,7 +490,6 @@ be_btree_node_alloc(const struct m0_be_btree *btree, struct m0_be_tx *tx)
 	node = (struct m0_be_bnode *)mem_alloc(btree, tx,
 					       sizeof(struct m0_be_bnode),
 					       M0_BITS(M0_BAP_NORMAL));
-	M0_ASSERT(node != NULL);	/* @todo: analyse return code */
```

Comment:
Please explain, why is this removed. The same for other changes.

Reply:
These checks (node != NULL) are redundant,  same check is done in mem_alloc, thus codacy was giving following error : 

> Either the condition 'node!=NULL' is redundant or there is a possible null pointer dereference: p_node.